### PR TITLE
Add RpcLogCategory enum

### PR DESCRIPTION
**Pull request by yojagad**
Part of work for https://github.com/Azure/azure-functions-host/issues/3117

### CODE REVIEW

File: src/proto/FunctionRpc.proto [hunk 1]
```diff
@@ -391,6 +394,32 @@ message RpcLog {
   string properties = 7;
 }
 
+message RpcSystemLog {
+  enum Level {
+    Information = 0;
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: All fields here apply for both system and invocation logs.  Possible to create a base type that can be resused for system logs and invocation logs? If that would be breaking change, we have to add this behind capabilities flags

Comment (pragnagopa) [reply src/proto/FunctionRpc.proto]: ping @yojagad

Comment (yojagad) [reply src/proto/FunctionRpc.proto]: @pragnagopa So, with the latest updates to https://github.com/Azure/azure-functions-nodejs-worker/pull/239 , I reused RpcLog as is (I'm using category to distinguish as you suggested). If that looks good, these set of proto changes may not be needed anymore and I can abandon this PR.

**Comment by pragnagopa** [comment]
Can you please introduce a new enum RpcLogCategory instead of reusing string category. That would be more explicit and out-of-proc langauges can have clear definition of a system log

**Comment by yojagad** [comment]
> RpcLogCategory
@pragnagopa Is the intention also to move startup, load, invocation, etc towards the enum rather than string?

**Comment by pragnagopa** [comment]
yes, idea is to move to enum so that host and language workers can look for specific values instead of relying on strings

**Comment by yojagad** [comment]
> > RpcLogCategory
> > @pragnagopa Is the intention also to move startup, load, invocation, etc towards the enum rather than string?

Added enum.

**Comment by pragnagopa** [comment]
delete RpcSystemLog  assuming it is not needed anymore

File: src/proto/FunctionRpc.proto [hunk 2]
```diff
@@ -389,6 +395,9 @@ message RpcLog {
 
   // json serialized property bag, or could use a type scheme like map<string, TypedData>
   string properties = 7;
+
+  // Category of the log. Either user(default) or system.
+  RpcLogCategory logCategory = 8;
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: change to RpcLogCategory log_category

File: src/proto/FunctionRpc.proto [hunk 3]
```diff
@@ -368,6 +368,12 @@ message RpcLog {
     None = 6;
   }
 
+  // Category of the log. Defaults to User if not specified.
```
Comment (mhoeger) [inline src/proto/FunctionRpc.proto]: nice!!

**Pull request merged**